src/main/java/com/openshift/internal/restclient/model/Pod.java

```diff
@@ -61,10 +74,38 @@ public String getHost() {
 		}
 		return images;
 	}
+	
+	/**
+	 * The logic of the method is a copied from 'podStatus' function of
+	 * [app/scripts/filters/resources.js] of [openshift/origin-web-console]
+	 */
 
 	@Override
 	public String getStatus() {
-		return asString(POD_STATUS);
+		if (get(POD_DELETION_TIMESTAMP).isDefined()) {
```

Comment:
@jcantrill 
Yes, i thought about the tests. 
The existing `PodTest` is actually going through all the loops over `containerStatuses`.
In order to test it properly we need to create more `v1_pod` jsons for every situation. I'm ok with it, but maybe you can offer a more nice option